editor/filesystem_dock.cpp

```diff
@@ -1326,6 +1326,15 @@ void FileSystemDock::_fs_changed() {
 	set_process(false);
 }
 
+void FileSystemDock::_directory_created(const String &p_path) {
+	if (!DirAccess::exists(p_path)) {
```

Comment:
I'm wondering if the `if (!DirAccess::exists(p_path)) {` could be done in `EditorFileSystem::add_new_directory` and that `EditorFileSystem::add_new_directory` emits the `filesystem_changed` signal instead of updating the tree directly here.
The point is that with the current implementation of `EditorFileSystem::add_new_directory`, it's not usable anywhere else in the Editor code and is not really "self contained".